pkgs/http/lib/src/client.dart

```diff
@@ -148,8 +148,10 @@ abstract interface class Client {
 
   /// Closes the client and cleans up any resources associated with it.
   ///
-  /// It's important to close each client when it's done being used; failing to
-  /// do so can cause the Dart process to hang.
+  /// Some clients may maintain a pool of network connections that will not
+  /// be disconnected until the client is closed. This may cause programs run
```

Comment:
[nit] We don't need to hedge twice with "Some" and "may". Also I think it reads better adjusting the tense in the middle of the first sentence.

```suggestion
  /// Some clients maintain a pool of network connections that are not
  /// disconnected until the client is closed. This may cause programs run
```